### PR TITLE
added JSONschema validation to labelbox

### PR DESCRIPTION
**Pull request by Fl4m3Ph03n1x**
Background
----

I have, for the first time in darwin-py's life, added JSON schema validation, this time to Labelbox. 
However, there are a couple of things I am not a big fan of:
- The error message for `oneOf` is simply wrong. It says `"point" is required` which is not true, `point` is only one of the possible valued (bbox and polygon being the others).
- After double checking, I have realized `jsonschema` is only compatible with Python 3.7 and onwards. I have to use an alternative. 

In the meantime I would like a review of the code. What do you guys think?

How can I test this?
----

Simply run `pytest`.

EDIT
----

Replaced `jsonschema` with `jsonschmea_rs` because the second one is the only one compatible with all the versions of Python `darwin-py` official supports. 

Although the issue with `oneOf` is now fixed (it does not exist in this library`, we have however another, perhaps more important issue:
- None of the missing required properties for internal objects belonging to `oneOf` are shown as missing. Instead we just get a general error "Could not match any of the schemas".

This is not really any better, one might even say the loss of granularity is so severe it can be even worst. However, it is literally the only one compatible with Python version 3.6 - 3.9.

We should probably discuss this.

### CODE REVIEW

File: darwin/importer/formats/labelbox_schemas.py
Comment (kamilc) [inline darwin/importer/formats/labelbox_schemas.py:48]: You don't need the `"type": "object"` here as it's already included in the level above.

File: darwin/importer/formats/labelbox_schemas.py
Comment (kamilc) [inline darwin/importer/formats/labelbox_schemas.py:73]: Are those defaults meaningful? Does it convey any real logic here?

File: darwin/importer/formats/labelbox_schemas.py
Comment (kamilc) [inline darwin/importer/formats/labelbox_schemas.py:12]: Could we add some constraints, like e. g.: `"minimum": 0`?

File: darwin/importer/formats/labelbox_schemas.py
Comment (kamilc) [inline darwin/importer/formats/labelbox_schemas.py:43]: Should it include `"$id"`?

File: darwin/importer/formats/labelbox.py
Comment (kamilc) [inline darwin/importer/formats/labelbox.py:92]: Would it look nicer to do `str(obj.get("title"))` instead of ignoring the types?

File: darwin/importer/formats/labelbox.py
Comment (kamilc) [inline darwin/importer/formats/labelbox.py:18]: I think that it now misses `schema` in the docstring.

**Review by kamilc — CHANGES_REQUESTED** [review]

File: darwin/importer/formats/labelbox_schemas.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox_schemas.py:48]: I am specifying that the items of the array of of type "object". The above "object" type is for the father object.
Unless there is an implicit rule I am not aware of, I think this is correct.

File: darwin/importer/formats/labelbox_schemas.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox_schemas.py:73]: Those defaults are shown when documentation tools are used to pre-fill test fields.

File: darwin/importer/formats/labelbox_schemas.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox_schemas.py:12]: I don;t know if we have any constraints for `top` and `left`. I think the `minimum` one is a good start for `height` and `width` though!

File: darwin/importer/formats/labelbox_schemas.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox_schemas.py:43]: I am not sure all objects here should have `$id`. Reality is I subdivided this into so many objects because I the structure was quiet complex and big and this made it easier to understand. The only object I am actually reusing is `point`.

We can  have a talk about this if you want.

File: darwin/importer/formats/labelbox.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox.py:92]: Although that would work for `str(None)` it would not for the other cases. 
If you have any suggestions, I am all ears. The only ones I found were:
https://stackoverflow.com/questions/68446642/how-do-i-get-pylance-to-ignore-the-possibility-of-none

Any of them sound better then ignoring type casting, but I have not settled on one. @andreaazzini  for example is not a fan of `assert`s.

File: darwin/importer/formats/labelbox.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox.py:18]: Correct!

File: darwin/importer/formats/labelbox_schemas.py
Comment (kamilc) [reply darwin/importer/formats/labelbox_schemas.py:48]: This isn't how it works. Whatever you specify under `oneOf` gets merged with the level above. So here the level above specifies `type: object` and alternatives under `oneOf` specify the same which result in... `type: object`. Let's take an example (pseudo-code for brevity):

```yaml
type: object
properties:
  title:
    type: string
oneOf:
  - properties:
      one:
        type: string
    required: one
  - properties:
      two:
        type: string
    required: two
```
Makes it resolve to these two alternative schemas:

```yaml
type: object
properties:
  title:
    type: string
  one:
    type: string
required: one
```
and:
```yaml
type: object
properties:
  title:
    type: string
  two:
    type: string
required: two
```

... which are tried one after another by the validator. Notice that adding `type: object` to children of `oneOf` wouldn't change the resulting alternative schemas at all.

Hope this helps.

File: darwin/importer/formats/labelbox_schemas.py
Comment (kamilc) [reply darwin/importer/formats/labelbox_schemas.py:12]: We most probably don't have constraints for top and left but yes, width and height can't really be negative.

File: darwin/importer/formats/labelbox.py
Comment (kamilc) [reply darwin/importer/formats/labelbox.py:92]: Isn't this `title` coming from the schema that has just been validated? If so, then we know for sure that it's a string-like variable. In this case, doing `str(...)` should be pretty safe. Am I not seeing something here?

File: darwin/importer/formats/labelbox_schemas.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox_schemas.py:48]: Fixed !

File: darwin/importer/formats/labelbox_schemas.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox_schemas.py:12]: Fixed !

File: darwin/importer/formats/labelbox.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox.py:92]: My comment is in regards to Pylance. Pylance doesnt know there was  a validation and therefore will always report an error. It is this I am trying to address here.

Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox.py:92]: Fixed is static `cast`

File: darwin/importer/formats/labelbox.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox.py:18]: Waiting for final approval before fixing the docs :D

**Review by kamilc — APPROVED** [review]

File: darwin/importer/formats/labelbox.py
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/labelbox.py]: Fixed

**Review by kamilc — APPROVED** [review]

**Pull request merged**
